src/Modules/DatePickerInput.react.js

```diff
@@ -1,302 +1,286 @@
-                        (index === 1 && !DatePickerUtils.isSameDay(this.props.dateEnd, newDate)))
-                    ) {
-                        this.props.onSelect(newDate, dateType);
-                    }
-                });
-            } else {
-                this.props.onSelect(null, null, true);
-            }
+    _getMoment(date, isDateRange) {
+        return {
+            date: isDateRange ? undefined : date || moment(),
+            dateFrom: !isDateRange ? undefined : date || moment(),
+            dateTo: !isDateRange ? undefined : date || moment(),
+        };
+    }
+
+    _onCalendarChange({ date, dateFrom, dateTo }) {
+        const { onChange } = this.props;
+
+        if (!_.isUndefined(onChange)) {
+            onChange({ date, dateFrom, dateTo });
+        } else {
+            const { locale } = this.props;
 
             this.setState({
-                maybeDateRange: {
-                    dateEnd: value ? datesArray[1].trim() : '',
-                    dateStart: value ? datesArray[0].trim() : ''
-                }
+                date,
+                inputValue: this._safeDateFormat(date, locale),
             });
-        } else if (uxMode === 'calendar' || uxMode === 'input' && type === 'singleDate') {
-            const date = moment(value, this._dateFormats, newLocale, true);
+        }
+    }
 
-            if (date.isValid() && !DatePickerUtils.isDayDisabled(date, this.props)) {
-                this.props.onSelect(date);
-            } else if (value === '' || value === '__/__/____') {
-                this.props.onSelect(null, null, true);
-            }
+    _onCalendarClickOutside(event) {
+        this._setOpen(false);
+    }
 
-            if (uxMode === 'input') {
-                this.setState({ maybeDate: value });
-            }
+    _onIconClick() {
+        ReactDOM.findDOMNode(this._datePickerInput._input).focus();
+    }
 
-            if (uxMode === 'calendar') {
-                this.setState({
-                    maybeDateRange: {
-                        dateEnd: dateType === 'dateEnd' ? value : this.state.maybeDateRange.dateEnd,
-                        dateStart: dateType === 'dateStart' ? value : this.state.maybeDateRange.dateStart
-                    }
-                });
-            }
-        }
+    _onInputBlur(event) {
+        const { onBlur } = this.props;
 
-        this.props.hasValue(value);
+        if (!_.isUndefined(onBlur)) {
+            onBlur(event);
+        }
     }
 
-    _onClick(dateField) {
-        if (_.isFunction(this.props.onClick)) {
-            this.props.onClick();
+    _onInputChange(value) {
+        const { locale } = this.props;
+        const date = moment(value, this._dateFormats, locale || moment.locale(), true);
+
+        if (date.isValid() && !DatePickerUtils.isDayDisabled(date, this.props)) {
+            this._onCalendarChange(date);
+        } else if (value === '' || value === '__/__/____') {
```

Comment:
I do wish we could prevent various components using an input mask from passing out values containing only the mask placeholder characters.  I've seen similar behavior from things like `<PhoneInput>` and I believe our production database contains a number of phone numbers with the value `+1 (___) ___-____` ... saved that way before we improved validation :-(  But thanks for at least catching it here and acting appropriately.